src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFCD.cs

```diff
@@ -226,79 +242,57 @@ public static string AnchorDescription(int anchors)
         }
 #endif
 
-        /*
-         * To avoid recursion, we use a simple integer stack.
-         * This is the push.
-         */
-        private void PushInt(int I)
+        /// <summary>
+        /// To avoid recursion, we use a simple integer stack.
+        /// </summary>
+        private void PushInt(int i)
         {
-            if (_intDepth >= _intStack.Length)
-            {
-                int[] expanded = new int[_intDepth * 2];
-
-                Array.Copy(_intStack, 0, expanded, 0, _intDepth);
-
-                _intStack = expanded;
-            }
-
-            _intStack[_intDepth++] = I;
+            _intStack.Append(i);
         }
 
-        /*
-         * True if the stack is empty.
-         */
         private bool IntIsEmpty()
         {
-            return _intDepth == 0;
+            return _intStack.Length == 0;
         }
 
-        /*
-         * This is the pop.
-         */
         private int PopInt()
         {
-            return _intStack[--_intDepth];
+            return _intStack.Pop();
         }
 
-        /*
-          * We also use a stack of RegexFC objects.
-          * This is the push.
-          */
+        /// <summary>
+        /// We also use a stack of RegexFC objects.
+        /// </summary>
         private void PushFC(RegexFC fc)
         {
-            if (_fcDepth >= _fcStack.Length)
-            {
-                RegexFC[] expanded = new RegexFC[_fcDepth * 2];
-
-                Array.Copy(_fcStack, 0, expanded, 0, _fcDepth);
-                _fcStack = expanded;
-            }
-
-            _fcStack[_fcDepth++] = fc;
+            _fcStack.Add(fc);
         }
 
-        /*
-         * True if the stack is empty.
-         */
         private bool FCIsEmpty()
         {
-            return _fcDepth == 0;
+            return _fcStack.Count == 0;
         }
 
-        /*
-         * This is the pop.
-         */
         private RegexFC PopFC()
         {
-            return _fcStack[--_fcDepth];
+            RegexFC item = TopFC();
+            _fcStack.RemoveAt(_fcStack.Count - 1);
+
+            return item;
         }
 
-        /*
-         * This is the top.
-         */
         private RegexFC TopFC()
         {
-            return _fcStack[_fcDepth - 1];
+            return _fcStack[_fcStack.Count - 1];
+        }
+
+        /// <summary>
+        /// Return rented buffers. Clear RegexFC buffer manually as only
```

Comment:
What code is the "Clear RegexFC buffer manually" referring to?

Reply:
thanks.